azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADOAuth2AutoConfiguration.java

```diff
@@ -22,18 +30,36 @@
 @PropertySource(value = "classpath:serviceEndpoints.properties")
 @EnableConfigurationProperties({AADAuthenticationProperties.class, ServiceEndpointsProperties.class})
 public class AADOAuth2AutoConfiguration {
-    private AADAuthenticationProperties aadAuthProps;
-    private ServiceEndpointsProperties serviceEndpointsProps;
+
+    private final TelemetryProxy telemetryProxy;
+
+    private final AADAuthenticationProperties aadAuthProps;
+
+    private final ServiceEndpointsProperties serviceEndpointsProps;
 
     public AADOAuth2AutoConfiguration(AADAuthenticationProperties aadAuthProperties,
-                                      ServiceEndpointsProperties serviceEndpointsProps) {
+                                      ServiceEndpointsProperties serviceEndpointsProps,
+                                      TelemetryProxy telemetryProxy) {
         this.aadAuthProps = aadAuthProperties;
         this.serviceEndpointsProps = serviceEndpointsProps;
+        this.telemetryProxy = telemetryProxy;
+
+        trackCustomEvent();
```

Comment:
How about we track the event in `@PostConstruct` instead of in constructor?

Reply:
make sense. will update it.